algorithms/src/Kokkos_Sort.hpp

```diff
@@ -527,8 +548,10 @@ struct min_max_functor {
 
 }  // namespace Impl
 
-template <class ViewType>
-void sort(ViewType const& view, bool const always_use_kokkos_sort = false) {
+template <class ExecutionSpace, class ViewType>
+std::enable_if_t<Kokkos::is_execution_space<ExecutionSpace>::value> sort(
+    const ExecutionSpace& exec, ViewType const& view,
+    bool const always_use_kokkos_sort = false) {
   if (!always_use_kokkos_sort) {
     if (Impl::try_std_sort(view)) return;
   }
```

Comment:
This shortcut becomes interesting when passing an execution space.  I need to think about it more.

Reply:
Yes, we can discuss this. In case, we want `Kokkos::sort` to behave as `deep_copy`, we should add a fence after `std::sort` so that the user can rely on the kernel being enqueued.

Reply:
Fixed in https://github.com/kokkos/kokkos/pull/4490/commits/77129cedd8ce67aa4cb9b801fd55dec1f8285911.